### ISSUE TITLE
Add support for RandomGenerator in Java 17

### ISSUE THREAD

**Issue by fabiolimace** [f4b6a3/ksuid-creator#13]

**Comment by fabiolimace** [comment]
Now it is easy to use `RandomGenerator` (JDK17) and `SplittableRandom` (JDK8).

Example:

```java
// use a random function that returns a long value
RandomGenerator random = RandomGenerator.getDefault();
KsuidFactory factory = KsuidFactory.newInstance(() -> random.nextLong());
```

**Comment by fabiolimace** [comment]
Released v4.0.0.
